libfreerdp/core/transport.c

```diff
@@ -371,6 +371,35 @@ BOOL transport_accept_nla(rdpTransport* transport)
 	return TRUE;
 }
 
+void transport_bio_error_log(char *biofunc, rdpTransport* transport)
+{
+	unsigned long sslerr;
+	char *buf;
+
+	if (!transport->frontBio)
+	{
+		WLog_ERR(TAG, "%s called with transport->frontBio null", biofunc);
+		return;
+	}
+
+	if (ERR_peek_error() == 0)
+	{
+		WLog_ERR(TAG, "%s returned system error %d: %s", biofunc, errno, strerror(errno));
+		return;
+	}
+
+	buf = malloc(120);
+	if (buf)
+	{
+		while((sslerr = ERR_get_error()))
+		{
+			ERR_error_string(sslerr, buf);
```

Comment:
I will add _n and 120 to accomodate @akallabeth paranoid programming style :)

I will also check how to move original `__FUNCTION__`, `__LINE__` and `__FILE__` values inside my function (not yet understood how).

I hope I will be able to do it by tomorrow.